tensorflow/core/kernels/string_split_op.cc

```diff
@@ -28,19 +28,62 @@ namespace tensorflow {
 
 namespace {
 
-std::vector<string> Split(const string& str, const string& delimiter,
-                          const bool skipEmpty) {
+Status Split(std::vector<string>& char_vector, const string& str,
+             const string& delimiter, const bool skipEmpty,
+             const string& encoding) {
   if (!delimiter.empty()) {
     if (skipEmpty) {
-      return str_util::Split(str, delimiter, str_util::SkipEmpty());
+      char_vector = str_util::Split(str, delimiter, str_util::SkipEmpty());
+      return Status::OK();
     }
-    return str_util::Split(str, delimiter);
+    char_vector = str_util::Split(str, delimiter);
+    return Status::OK();
   }
-  std::vector<string> char_vector(str.size());
-  for (size_t i = 0; i < str.size(); ++i) {
-    char_vector[i] = str[i];
+  if (encoding == "utf8") {
+    // Bytes    Byte 1    Byte 2    Byte 3    Byte 4
+    //   1     0xxxxxxx
+    //   2     110xxxxx  10xxxxxx
+    //   3     1110xxxx  10xxxxxx  10xxxxxx
+    //   4     11110xxx  10xxxxxx  10xxxxxx  10xxxxxx
+    char_vector.reserve(str.size());
+    size_t off = 0, len = 0;
+    for (size_t i = 0; i < str.size(); ++i) {
+      if (off == i) {
+        if ((str[i] & 0x80) == 0x00) {
+          char_vector.push_back(str.substr(off, 1));
+          off++;
+        } else if ((str[i] & 0xE0) == 0xC0) {
+          len = 2;
+        } else if ((str[i] & 0xF0) == 0xE0) {
+          len = 3;
+        } else if ((str[i] & 0xF8) == 0xF0) {
+          len = 4;
+        } else {
+          char_vector.clear();
+          return errors::Internal("invalid utf8 encoding");
+        }
+      } else {
+        if ((str[i] & 0xC0) != 0x80) {
+          char_vector.clear();
+          return errors::Internal("invalid utf8 encoding");
+        }
+        if (off + len == i + 1) {
+          char_vector.push_back(str.substr(off, len));
+          off += len;
+        }
+      }
+    }
+    if (off < str.size()) {
+      // Not enough chars
```

Comment:
Remove comment and put that in the error message.

Reply:
Done.